ipykernel/zmqshell.py

```diff
@@ -70,6 +78,15 @@ def _flush_streams(self):
         sys.stdout.flush()
         sys.stderr.flush()
 
+    @default('thread_local')
```

Comment:
Master is already at 4.2 - 

https://github.com/ipython/ipywidgets/blob/bb5092be3c0e75a12293472584a218d9a217c935/setup.py#L108

so i'll just merge on that.

Thanks!